homeassistant/components/launch_library/sensor.py

```diff
@@ -1,69 +1,83 @@
-"""A sensor platform that give you information about the next space launch."""
+"""Support for Launch Library sensors."""
 from __future__ import annotations
 
-from datetime import timedelta
-import logging
+from typing import Any
 
-from pylaunches import PyLaunches, PyLaunchesException
-import voluptuous as vol
+from pylaunches.objects.launch import Launch
 
-from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
-from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
 
 from .const import (
     ATTR_AGENCY,
     ATTR_AGENCY_COUNTRY_CODE,
     ATTR_LAUNCH_TIME,
     ATTR_STREAM,
     ATTRIBUTION,
-    DEFAULT_NAME,
+    DOMAIN,
 )
 
-_LOGGER = logging.getLogger(__name__)
 
-SCAN_INTERVAL = timedelta(hours=1)
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string}
-)
+    coordinator = hass.data[DOMAIN]
+
+    async_add_entities(
+        [
+            NextLaunchSensor(coordinator, entry.entry_id),
+        ]
+    )
+
+
+class LLBaseEntity(CoordinatorEntity, SensorEntity):
+    """Sensor base entity."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator) -> None:
+        """Initialize a Launch Library entity."""
+        super().__init__(coordinator)
+
+    def get_next_launch(self) -> Launch | None:
+        """Return next launch."""
+        return next((launch for launch in self.coordinator.data), None)
+
+
+class NextLaunchSensor(LLBaseEntity):
+    """Representation of the next launch sensor."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_icon = "mdi:rocket-launch"
+    _attr_name = "Next launch"
+
+    def __init__(self, coordinator: DataUpdateCoordinator, entry_id: str) -> None:
+        """Create Next launch sensor from base entity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{entry_id}_next_launch"
```

Comment:
Are there a better way to do this? I want to make it flexible enough so easily can add other sensors I have planned

Reply:
Yes, by using EntotyDescription and the `key` attribute of those.
But even in that case, it would need some more work to be dynamic since the attributes for this sensor class are for the next launch.
So let's look at that for another PR.